RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,318 @@
+    return &r;
+  }
+
+  inline pointer allocate (size_type n) {
+    void * p = nullptr;
+    posix_memalign(&p,N,n*sizeof(value_type));
+    return (pointer)p;
+  }
+
+  inline void deallocate (pointer p, size_type) {
+    free(p);
+  }
+
+  inline void construct (pointer p, const value_type & wert) {
+     new (p) value_type (wert);
+  }
+
+  inline void destroy (pointer p) {
+    p->~value_type ();
+  }
+
+  inline size_type max_size () const throw () {
+    return size_type (-1) / sizeof (value_type);
+  }
+
+  template <typename T2>
+  struct rebind {
+    typedef AlignmentAllocator<T2, N> other;
+  };
+
+  bool operator!=(const AlignmentAllocator<T,N>& other) const  {
+    return !(*this == other);
+  }
+
+  // Returns true if and only if storage allocated from *this
+  // can be deallocated from other, and vice versa.
+  // Always returns true for stateless allocators.
+  bool operator==(const AlignmentAllocator<T,N>& other) const {
+    return true;
+  }
+};
+
+
+class DAClusterizerInZT_vect  final : public TrackClusterizerInZ {
+
+public:
+  template<typename T> 
+  using AlignedVector = std::vector<T, AlignmentAllocator<T, 16> >;
+
+  // Internal data structure to 
+  struct track_t {
+    
+    void AddItem( double new_z, double new_t, double new_dz2, double new_dt2, const reco::TransientTrack* new_tt, double new_pi   )
+    {
+      z.push_back( new_z );
+      t.push_back( new_t );
+      dz2.push_back( new_dz2 );
+      dt2.push_back( new_dt2 );
+      errsum.push_back( 1./(1./new_dz2 + 1./new_dt2) );
+      tt.push_back( new_tt );
+      
+      pi.push_back( new_pi ); // track weight
+      Z_sum.push_back( 1.0 ); // Z[i]   for DA clustering, initial value as done in ::fill
+    }
+
+    
+    
+    unsigned int GetSize() const
+    {
+      return z.size();
+    }
+    
+    
+    // has to be called everytime the items are modified
+    void ExtractRaw()
+    {
+      _z = &z.front();
+      _t = &t.front();
+      _dz2 = &dz2.front();
+      _dt2 = &dt2.front();
+      _errsum = &errsum.front();
+      _Z_sum = &Z_sum.front();
+      _pi = &pi.front();
+    }
+    
+    double * __restrict__ _z __attribute__ ((aligned (16))); // z-coordinate at point of closest approach to the beamline
+    double * __restrict__ _t __attribute__ ((aligned (16))); // t-coordinate at point of closest approach to the beamline
+    double * __restrict__ _dz2 __attribute__ ((aligned (16))); // square of the error of z(pca)
+    double * __restrict__ _dt2 __attribute__ ((aligned (16))); // square of the error of t(pca)
+    double * __restrict__ _errsum __attribute__ ((aligned (16))); // sum of squares of the pca errors
+    
+    double * __restrict__  _Z_sum __attribute__ ((aligned (16))); // Z[i]   for DA clustering
+    double * __restrict__  _pi __attribute__ ((aligned (16))); // track weight
+    
+    AlignedVector<double> z; // z-coordinate at point of closest approach to the beamline
```

Comment:
Is there a reason you prefer to have a struct of vectors, rather than a single vector of a struct with all the necessary (singleton) member variables? (for both track and vertex)